### PR TITLE
Remove databank owners form [#179685221]

### PR DESCRIPTION
**Pull request by olzraiti**
Currently the list of the data bank forms is the `$.options.forms` list of form `MHL.67`. This causes a bug that when navigating to edit a document of some databank form that is on that list, the URL get malformed as the project-form code interprets the data bank form being a sub form, which it isn't in this context.

To fix this, the list of the data bank forms should be derived from filtering all the forms with the `$.options.databank` option. The form with list of the data bank forms (`MHL.67`) should be removed after this.

### CODE REVIEW

**Comment by olzraiti** [comment]
WIP! This doesn't work, could somebody help with the RXJS stuff? I tried two implementations for the `this.forms$` pipe in `datasets.component.ts`. Both work for ICT admin user, but for a regular user no forms show up.

File: projects/laji/src/app/+theme/datasets/datasets.component.ts [hunk 1]
```diff
@@ -34,25 +34,33 @@ export class DatasetsComponent {
...
-            concatMap(person => UserService.isIctAdmin(person) ? of(f) : this.formPermissionService.hasAccessToForm(f && f.id || null).pipe(
-              map((access) => access ? f : null)
-            ))
-          )),
+    //Doesn't work!
+    this.forms$ = this.formService.getAllForms().pipe(
+      switchMap(fs => forkJoin(
+        ...fs.filter(f =>
+          f.options?.dataset && ![Global.forms.datasetPrimary, Global.forms.datasetSecondary].includes(f.id)
+        ).map(f => this.formPermissionService.getRights(f).pipe(
```
Comment (olzraiti) [inline projects/laji/src/app/+theme/datasets/datasets.component.ts:42]: For a regular user, `this.formPermissionService.getRights(f)` fires only once. For ICT admin it works.

File: projects/laji/src/app/+theme/datasets/datasets.component.ts [hunk 2]
```diff
@@ -34,25 +34,33 @@ export class DatasetsComponent {
...
+    );
+    //Doesn't also work!
+    // this.forms$ = combineLatest(this.userService.user$, this.formService.getAllForms()).pipe(
+    //  switchMap(([person, fs]) => forkJoin(
+    //    ...fs
+    //      .filter(f =>
+    //        f.options?.dataset && ![Global.forms.datasetPrimary, Global.forms.datasetSecondary].includes(f.id)
+    //      ).map(f => (UserService.isIctAdmin(person)
+    //          ? of(true)
+    //          : this.formPermissionService.hasAccessToForm(f.id)
```
Comment (olzraiti) [inline projects/laji/src/app/+theme/datasets/datasets.component.ts:56]: For a regular user, `this.formPermissionService.hasAccessToForm(f.id)` fires only once. For ICT admin it works.

**Comment by olzraiti** [comment]
Fixed. OK for review

**Review by aorin — APPROVED** [review]
:+1:

**Pull request merged**